erizo/src/erizo/rtp/BandwidthEstimationHandler.cpp

```diff
@@ -98,24 +98,203 @@ void BandwidthEstimationHandler::updateExtensionMap(bool is_video, std::array<RT
   }
 }
 
+// Implements x < y, taking into account RTP sequence number wrap
+// The general idea is if there's a very large difference between
+// x and y, that implies that the larger one is actually "less than"
+// the smaller one.
+//
+// I picked 0x8000 as my "very large" threshold because it splits
+// 0xffff, so it seems like a logical choice.
+bool BandwidthEstimationHandler::rtpSequenceLessThan(uint16_t x, uint16_t y) {
+  int diff = y - x;
+  if (diff > 0) {
+    return (diff < 0x8000);
+  } else if (diff < 0) {
+    return (diff < -0x8000);
+  } else {  // diff == 0
+    return false;
+  }
+}
+
 void BandwidthEstimationHandler::read(Context *ctx, std::shared_ptr<dataPacket> packet) {
   temp_ctx_ = ctx;
   if (!running_) {
     process();
     running_ = true;
   }
-  RtcpHeader *chead = reinterpret_cast<RtcpHeader*> (packet->data);
-  if (!chead->isRtcp() && packet->type == VIDEO_PACKET) {
+
+  RtcpHeader *chead = reinterpret_cast<RtcpHeader*>(packet->data);
+  RtpHeader *head = reinterpret_cast<RtpHeader*>(packet->data);
+  uint32_t now = ClockUtils::timePointToMs(clock::now());
+
+  if (!chead->isRtcp()) {
+    // RTP PACKET
     if (parsePacket(packet)) {
+      webrtc::StatisticianMap res = _stats_handler->GetActiveStatisticians();   // get a list of active stats engine
+      webrtc::StatisticianMap::iterator it;
+
       int64_t arrival_time_ms = packet->received_time_ms;
       arrival_time_ms = clock_->TimeInMilliseconds() - (ClockUtils::timePointToMs(clock::now()) - arrival_time_ms);
       size_t payload_size = packet->length;
-      pickEstimatorFromHeader();
-      rbe_->IncomingPacket(arrival_time_ms, payload_size, header_);
+      if (packet->type == VIDEO_PACKET) {
+        // SEND TO BWE
+        pickEstimatorFromHeader();
+        rbe_->IncomingPacket(arrival_time_ms, payload_size, header_);
+
+        // SEND TO OUR STATS ENGINE
```

Comment:
remove comments like these ones in general please, I don't think they give us more info